<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1

In [ ]:
while True:
    print('*', end='')

# Exercise 2

In [1]:
print('Hello World')

Hello World


# Exercise 3

In [14]:
%%timeit
x=10
y=10
z=10


squares = [i**2 for i in range(10)]



302 ns ± 6.87 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [15]:
%%timeit

squares = [i**2 for i in range(10)]

295 ns ± 7.72 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [16]:
squares = [i**2 for i in range(10)]

In [13]:
%%timeit

squares = list()
for i in range(10):
    squares.append(i**2)



372 ns ± 1.09 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


Taking "302 ns ± 6.87 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)" as an example, the first number 302ns is the mean time it takes for a loop of the cell to be executed i.e. run all the code in a cell. 6.87ns is the standard deviation. These values are from the best(shortest time) of 7 runs of 1 million loops executed each run.


## Footnotes